File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        double bad = 0;
        double good = x;
        while (good - bad > 0.00001) {
            double mid = bad + (good - bad) / 2;
            if (mid * mid >= x) {
                good = mid;
            } else {
                bad = mid;
            }
        }
        
        return (int) good;
    }
}

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        bad = 0
        good = x
        for _ in range(50):
            mid = (bad + good) / 2
            if mid * mid >= x:
                good = mid
            else:
                bad = mid
        return int(good)

        